tests/integration/recipe/test_recipe.py

```diff
@@ -2323,731 +2104,32 @@ def test_landmask_no_fx(tmp_path, patched_failing_datafinder, session):
-
-    # Check generated tasks
-    assert len(recipe.tasks) == 1
-    task = recipe.tasks.pop()
-    assert task.name == 'diagnostic_name' + TASKSEP + 'tas'
-    assert len(task.products) == 1
-    product = task.products.pop()
-
-    # Check volume_statistics
-    assert 'volume_statistics' in product.settings
-    settings = product.settings['volume_statistics']
-    assert len(settings) == 1
-    assert settings['operator'] == 'mean'
-
-    # Check legacy method of adding supplementary variables
-    assert len(product.datasets) == 1
-    dataset = product.datasets[0]
-    supplementaries = {
-        ds.facets['short_name']: ds
-        for ds in dataset.supplementaries
-    }
-    assert len(list(supplementaries)) == 2
-    sftgif_ds = supplementaries['sftgif']
-    assert sftgif_ds.facets['mip'] == 'fx'
-    volcello_ds = supplementaries['volcello']
-    assert volcello_ds.facets['ensemble'] == 'r2i1p1f1'
-    assert volcello_ds.facets['mip'] == 'Ofx'
-
-
-def test_fx_vars_invalid_mip_cmip6(tmp_path, patched_datafinder, session):
-    """Test fx variables with invalid mip."""
-    TAGS.set_tag_values(TAGS_FOR_TESTING)
-
-    content = dedent("""
-        preprocessors:
-          preproc:
-           area_statistics:
-             operator: mean
-             fx_variables:
-               areacella:
-                 mip: INVALID
-
-        diagnostics:
-          diagnostic_name:
-            variables:
-              tas:
-                preprocessor: preproc
-                project: CMIP6
-                mip: Amon
-                exp: historical
-                start_year: 2000
-                end_year: 2005
-                ensemble: r1i1p1f1
-                grid: gn
-                additional_datasets:
-                  - {dataset: CanESM5}
-            scripts: null
-        """)
-    msg = ("Unable to load CMOR table (project) 'CMIP6' for variable "
-           "'areacella' with mip 'INVALID'")
-    with pytest.raises(RecipeError) as rec_err_exp:
-        get_recipe(tmp_path, content, session)
-    assert str(rec_err_exp.value) == INITIALIZATION_ERROR_MSG
-    assert msg in rec_err_exp.value.failed_tasks[0].message
-
-
-def test_fx_vars_invalid_mip_for_var_cmip6(tmp_path, patched_datafinder,
-                                           session):
-    """Test fx variables with invalid mip for variable."""
-    TAGS.set_tag_values(TAGS_FOR_TESTING)
-
+def test_wrong_project(tmp_path, patched_datafinder, session):
     content = dedent("""
         preprocessors:
           preproc:
-           area_statistics:
+           volume_statistics:
              operator: mean
-             fx_variables:
-               areacella:
-                 mip: Lmon
-
         diagnostics:
           diagnostic_name:
             variables:
-              tas:
+              tos:
                 preprocessor: preproc
-                project: CMIP6
-                mip: Amon
+                project: CMIP7
```

Comment:
:rofl: this will come back and bite you very soon :grin:

Reply:
Haha, yes, that's what I thought. It wasn't me who wrote that test.